include/plugin/runtime-support.h

```diff
@@ -294,17 +307,19 @@ void protocol_data_in(const hilti::rt::Bool& is_orig, const hilti::rt::Bytes& da
  * @param is_orig true to signal gap to originator side, false for responder
  * @param offset of the gap inside the protocol stream
  * @param length of the gap
+ * @param h optional handle to the child analyzer to signal a gap to
  */
 void protocol_gap(const hilti::rt::Bool& is_orig, const hilti::rt::integer::safe<uint64_t>& offset,
-                  const hilti::rt::integer::safe<uint64_t>& len);
+                  const hilti::rt::integer::safe<uint64_t>& len, const std::optional<ProtocolHandle>& h);
 
 /**
  * Signals EOD to all previously instantiated Zeek-side child protocol
  * analyzers and removes them.
  *
  * @param analyzer ID of analyzer previously instantiated through `protocol_begin()`.
+ * @param h optional handle to the child analyzer to terminate.
  */
-void protocol_end();
+void protocol_end(const std::optional<ProtocolHandle>& h);
```

Comment:
In addition to wanting to make the handle type opaque I am using a dedicated type of `ProtocolHandle` instead of a `std::optional<integer::safe<uint64_t>>` here since otherwise I run into issues in generated code where we have issue coercing `Null` to the optional safe integer (two conversions). This is similar to the following code in Spicy:

```cpp
std::optional<integer::safe<uint64_t>> foo(std::optional<integer::safe<uint64_t>> s) { return s; }
foo(hilti::rt::Null());
```